File: BEGINNER_MACHINE_LEARNING/Machine Learning A-Z/Part 1 - Data Preprocessing/missing_data.R
dataset =read.csv('MissingData.csv')
dataset$Age = ifelse(is.na(dataset$Age),ave(dataset$Age,FUN= function(x) mean(x,na.rm = TRUE)),dataset$Age)
dataset$Salary= ifelse(is.na(dataset$Salary),ave(dataset$Salary,FUN=function(x) mean(x,na.rm = TRUE)),dataset$Salary)
#View(dataset$Salary)
print(dataset[1:3])

File: BEGINNER_MACHINE_LEARNING/Machine Learning A-Z/Part 1 - Data Preprocessing/missing_data.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.preprocessing import Imputer
from  sklearn.preprocessing import LabelEncoder

def missingDatafinder():
    # data set read
    dataset = pd.read_csv('missingData.csv')
    xMatrix=dataset.iloc[:,:-1].values
    yMatrix =dataset.iloc[:,3].values
    #print(yMatrix)
    print(xMatrix)
    imputer =Imputer(missing_values='NaN',strategy='mean',axis=0)
    print(imputer)
    imputer.fit(xMatrix[:,1:3])
    print(imputer)
    xMatrix[:,1:3]=imputer.transform(xMatrix[:,1:3])
    print(xMatrix)





def main():
    missingDatafinder()
    return 0


if __name__ == '__main__':
    main()
